eval/src/vespa/eval/instruction/generic_merge.cpp

```diff
@@ -140,6 +172,10 @@ GenericMerge::make_instruction(const ValueType &lhs_type, const ValueType &rhs_t
                                const ValueBuilderFactory &factory, Stash &stash)
 {
     const auto &param = stash.create<MergeParam>(lhs_type, rhs_type, function, factory);
+    if (param.dense_subspace_size == 1) {
```

Comment:
I would prefer sending param to the Select function and having that choose the implementation (like for join)